lib/scope.js

```diff
@@ -146,7 +146,10 @@ module.exports = {
     let end_url = await scope.page.url();
     let navigated = start_url !== end_url;
     // Might be able to handle this in `scope.afterStep`
-    if ( navigated ) {  // if (winner.constructor.name === 'ElementHandle') {
+    if ( navigated ) {
+      // Save the id that comes next
+      let { question_has_id, id, id_matches } = await scope.examinePageID( scope, 'none to match' );
+      scope.report[ scope.scenario_name ].ids.push( id );
```

Comment:
We're using this `scope.report[ scope.scenario_name ].ids.push( id )` syntax in several places, and I can see us adding it to even more functions in the future. I think this will be more flexible if we replace this with a function `scope.addToReport(msg)` where add to report handles `scope.report[ scope.scenario_name ].ids.push( id )` (or whatever other implementation we end up with. That makes sure we don't repeat the implementation details in several places, and the implementation is easy to change whenever we need to. 

Not a blocker for now, but it's a nice-to-have change.

Reply:
I like that idea, though we may not want it to be simply for pushing ids as we may want to be able to report other things as well, so we may want to think about its interface a bit more. Made this one into an issue too.